### PR TITLE
Fix api port

### PR DESCRIPTION
**Pull request by jichenjc**
The kustomize will remove the # comments ... so no one knows the info provided
and even if we some one failed to set the failure, we should not let container panic ...
so I added a default value here

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: controllers/openstackcluster_controller.go [hunk 1]
```diff
@@ -170,10 +170,16 @@ func (r *OpenStackClusterReconciler) reconcileCluster(logger logr.Logger, cluste
 			return reconcile.Result{}, errors.Errorf("failed to get control plane machine: %v", err)
 		}
 		if controlPlaneMachine != nil {
+			var apiPort int
```
Comment (sbueringer) [inline controllers/openstackcluster_controller.go]: wouldn't this be place to default to 6443? If I read it correctly we're currently defaulting to `0` :)

File: examples/cluster/cluster.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ spec:
     # * Disable the apiServerPort property
     # single-node control-plane:
     # * Enable the apiServerPort property
-    #apiServerPort: 6443
+    apiServerPort: <disable when multi-node control-plane>
```
Comment (sbueringer) [inline examples/cluster/cluster.yaml]: The rest of the example is commented to support multi-node control plane per default. That's the only property the other way around after this PR. So i would suggest keeping the `#` at the start of the line

Comment (jichenjc) [reply examples/cluster/cluster.yaml]: if you take a look at generated result...
the # will be removed by my kustomize and that's the reason why I have the nil setting and 
it lead to panic... (see my changes in the go file)

see my output. all the things marked as comments are removed, guess it's not my kustomize version issue? I am using 3.10 already
```
apiVersion: cluster.x-k8s.io/v1alpha2
kind: Cluster
metadata:
  name: test-cluster
  namespace: test-cluster
spec:
  clusterNetwork:
    apiServerPort: <disable when multi-node control-plane>
    pods:
      cidrBlocks:
      - 192.168.0.0/16
    serviceDomain: cluster.local
    services:
      cidrBlocks:
      - 10.96.0.0/12
  infrastructureRef:
    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
    kind: OpenStackCluster
    name: test-cluster
    namespace: test-cluster
---
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
kind: OpenStackCluster
metadata:
  name: test-cluster
  namespace: test-cluster
spec:
  apiServerLoadBalancerAdditionalPorts:
  - 22
  apiServerLoadBalancerFloatingIP: <loadbalancer floating ip>
  apiServerLoadBalancerPort: 6443
  cloudName: openstack
  cloudsSecret:
    name: cloud-config
    namespace: test-cluster
  disablePortSecurity: true
  disableServerTags: true
  dnsNameservers: []
  externalNetworkId: <external-network-id>
```

File: controllers/openstackcluster_controller.go (hunk 1 above)
Comment (jichenjc) [reply controllers/openstackcluster_controller.go]: I forgot to set 6443 here...

File: examples/cluster/cluster.yaml (hunk 2 above)
Comment (sbueringer) [reply examples/cluster/cluster.yaml]: Yup the comments are removed through kustomize that's right. So they are not that useful right now. We can add the property for now. I think I would open another PR sooner or later to create separate multi-node & single-node examples with aproprate properties

**Pull request merged**